### PR TITLE
salieri: Support read-only mode if archive is already opened

### PR DESCRIPTION
**Pull request by Thog**
This improves shader cache resilience when people opens another program that touch the cache.zip.

### CODE REVIEW

**Review by riperiperi — APPROVED** [review]
Nice one, this will be useful for testing with multiple instances.

File: Ryujinx.Graphics.Gpu/Shader/Cache/CacheCollection.cs [hunk 1]
```diff
@@ -308,6 +318,20 @@ public void EnsureArchiveUpToDate()
...
             string archivePath = GetArchivePath();
 
+            if (IsReadOnly)
+            {
+                Logger.Warning?.Print(LogClass.Gpu, $"Cache collection archive in read-only, archiving task skipped.");
+
+                return;
+            }
+
+            if (CacheHelper.IsArchiveReadOnly(archivePath))
```
Comment (gdkchan) [inline Ryujinx.Graphics.Gpu/Shader/Cache/CacheCollection.cs:328]: Whats the difference between this check and the one above?

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

File: Ryujinx.Graphics.Gpu/Shader/Cache/CacheCollection.cs (hunk 1 above)
Comment (Thog) [reply Ryujinx.Graphics.Gpu/Shader/Cache/CacheCollection.cs:328]: This one ensure the cache wasn't locked later on (by another instance running) while exiting the first instance.
I also wanted to have a different warning for this to be more explicit for the user.

**Review by AcK77 — APPROVED** [review]

**Pull request merged**